### ISSUE TITLE
When fields are no longer inherited, they're removed and their DB column is dropped

### ISSUE THREAD

**Issue by luisg123v** [odoo/odoo#29820]
Impacted versions:
- 12.0

Steps to reproduce:
- Inherit any field from any existing model from a new module, e.g. res.partner's ref:
  ```
  class ResPartner(models.Model):
      _inherit = 'res.partner'
      ref = fields.Char(string='Other Reference')
  ```
- Update your module
- Remove the field declaration from your module
- Update again

Current behavior:
The field is removed and the database column is dropped once the field is no longer inherited.

Expected behavior:
The field shouldn't be removed because it's not defined in your module, just inherited.

Video/Screenshot link (optional):
[Here](https://youtu.be/33b8mbyVm1o) is a video showing the issue.

**Comment by luisg123v** [comment]
CC @nhomar  @yaniaular

**Comment by pedrobaeza** [comment]
Are you sure about that? This seems weird as there's a counter for knowing the references and the column is not dropped if the reference count is > 0.

**Comment by luisg123v** [comment]
@pedrobaeza  
Yes, I'm pretty sure. I'm experiencing this issue with several fields, and I tested the same example I provided to be sure.

For some reason, both field's external IDs are being removed, the one defined in the base module and the other one defined in my module.

**Comment by luisg123v** [comment]
opw-1921778

CC @moylop260

**Comment by luisg123v** [comment]
@mart-e 
I think this one is kind of dangerous. Whould you check, please?

@pedrobaeza & @moylop260 
Digging it a little, it seems that there's no counter for references. If it was previously referenced from the updated module, but it's no more being referenced by it, it's removed, despite if it's referenced from any other module:
https://github.com/odoo/odoo/blob/652cecffdebfb09c0b96f7aabf9711a30f4d3bcd/odoo/addons/base/models/ir_model.py#L1620-L1630

**Comment by mart-e** [comment]
Thanks for the report, it was fixed at 91882dd97238a006ef7
